src/commands/cmd_geo.cc

```diff
@@ -355,6 +363,249 @@ class CommandGeoRadius : public CommandGeoBase {
+
+        member_ = GET_OR_RET(parser.TakeStr());
+      } else if (parser.EatEqICase("fromlonlat")) {
+        auto s = setOriginType(kLongLat);
+        if (!s.IsOK()) return s;
+
+        longitude_ = GET_OR_RET(parser.TakeFloat());
+        latitude_ = GET_OR_RET(parser.TakeFloat());
+        s = ValdiateLongLat(&longitude_, &latitude_);
+        if (!s.IsOK()) return s;
+      } else if (parser.EatEqICase("byradius")) {
+        auto s = setShapeType(CIRCULAR);
+        if (!s.IsOK()) return s;
+        radius_ = GET_OR_RET(parser.TakeFloat());
+        std::string distance_raw = GET_OR_RET(parser.TakeStr());
+        s = ParseDistanceUnit(distance_raw);
+        if (!s.IsOK()) return s;
+      } else if (parser.EatEqICase("bybox")) {
+        auto s = setShapeType(RECTANGULAR);
+        if (!s.IsOK()) return s;
+        width_ = GET_OR_RET(parser.TakeFloat());
+        height_ = GET_OR_RET(parser.TakeFloat());
+        std::string distance_raw = GET_OR_RET(parser.TakeStr());
+        s = ParseDistanceUnit(distance_raw);
+        if (!s.IsOK()) return s;
+      } else if (parser.EatEqICase("asc") && sort_ == kSortNone) {
+        sort_ = kSortASC;
+      } else if (parser.EatEqICase("desc") && sort_ == kSortNone) {
+        sort_ = kSortDESC;
+      } else if (parser.EatEqICase("count")) {
+        count_ = GET_OR_RET(parser.TakeInt<int>(NumericRange<int>{1, std::numeric_limits<int>::max()}));
+      } else if (parser.EatEqICase("withcoord")) {
+        with_coord_ = true;
+      } else if (parser.EatEqICase("withdist")) {
+        with_dist_ = true;
+      } else if (parser.EatEqICase("withhash")) {
+        with_hash_ = true;
+      } else {
+        return {Status::RedisParseErr, "Invalid argument given"};
+      }
+    }
+
+    if (member_ != "" && longitude_ != 0 && latitude_ != 0) {
+      return {Status::RedisParseErr, "please use only one of FROMMEMBER or FROMLONLAT"};
+    }
+
+    auto s = createGeoShape();
+    if (!s.IsOK()) {
+      return s;
+    }
+    return Commander::Parse(args);
+  }
+
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    std::vector<GeoPoint> geo_points;
+    redis::Geo geo_db(svr->storage, conn->GetNamespace());
+
+    auto s = geo_db.Search(args_[1], geo_shape_, origin_point_type_, member_, count_, sort_, store_key_, false,
+                           GetUnitConversion(), &geo_points);
+
+    if (!s.ok()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    *output = generateOutput(geo_points);
+    // storing comes later.
+    return Status::OK();
+  }
+
+ protected:
+  double radius_ = 0;
+  double height_ = 0;
+  double width_ = 0;
+  int count_ = 0;
+  double longitude_ = 0;
+  double latitude_ = 0;
+  std::string member_;
+  std::string key_;
+  DistanceSort sort_ = kSortNone;
+  GeoShapeType shape_type_ = NONE;
+  OriginPointType origin_point_type_ = kNone;
+  GeoShape geo_shape_;
+  std::string store_key_;
+  bool store_distance_ = false;
```

Comment:
Actually, you don't use the `store_key_` variable in the `GEOSEARCH` command so you can delete it and use just `""` as a function parameter later. But this leads us to the situation when the `Geo::Search` function in the case of the `GEOSEARCH` command doesn't use 2 input parameters so maybe we can have two separate functions instead of one `Geo::Search`.

Reply:
Hmmm, agreed. One could be a simple search and one could be search store. Refactoring this.

Reply:
Hmmm, agreed. One could be a simple search and one could be search store. Refactoring this.